src/main/java/lotes/parser/Parser.java

```diff
@@ -0,0 +1,107 @@
+package lotes.parser;
+
+import java.util.Scanner;
+
+import lotes.storage.Storage;
+import lotes.task.TaskList;
+
+public class Parser {
+
+    // Parses the user input to get the command to perform.
+
+    public static String[] parseInputCommand(String userInput) {
+        String[] words = userInput.split(" ", 2);
+
+        return words;
+    }
+
+    // Reads and parse the user input to perform its respective actions.
+
+    public static boolean performUserInput(String userInput, TaskList taskList) throws LotesException {
+        boolean isExit = false;
+        String[] command = parseInputCommand(userInput);
+
+        switch (command[0]) {
+        case "bye":
+            System.out.println(TaskList.goodbyeMessage);
+            isExit = true;
+            break;
+        case "list":
+            taskList.printTasksList();
+            break;
+        case "mark":
+            taskList.markTask(userInput);
+            break;
+        case "unmark":
+            taskList.unMarkTask(userInput);
+            break;
+        case "todo":
+            taskList.addToDo(userInput);
+            break;
+        case "deadline":
+            taskList.addDeadline(userInput);
+            break;
+        case "event":
+            taskList.addEvent(userInput);
+            break;
+        case "delete":
+            taskList.deleteTask(userInput);
+            break;
+        case "add":
+            taskList.addNewTask(userInput);
+            break;
+        default:
+            if(command.length < 2) {
+                throw new LotesException();
+            }
+        }
+
+        if (command[0].equals("todo") || command[0].equals("deadline") || command[0].equals("event")
+        || command[0].equals("add") || command[0].equals("mark") || command[0].equals("unmark")
+        || command[0].equals("delete")) {
+            Storage.updateFile();
+        }
+
+        return isExit;
+    }
+
+    // Handle errors by catching exceptions.
+
+    public static boolean isException(String userInput, TaskList taskList) {
+        try {
+            if (Parser.performUserInput(userInput, taskList)) { // if ixExit returns true
+                return true;
+            }
+
+        } catch (NumberFormatException e) {
```

Comment:
Naming and comments are a bit strange.
You are actually executing the command here and catching the exception if one occurs.
However, the name gives an impression of a simple boolean check